operator/pkg/render/manifest.go

```diff
@@ -0,0 +1,371 @@
+	"istio.io/istio/pkg/kube"
+	pkgversion "istio.io/istio/pkg/version"
+)
+
+// GenerateManifest produces fully rendered Kubernetes objects from rendering Helm charts.
+// Inputs can be files and --set strings.
+// Client is option; if it is provided, cluster-specific settings can be auto-detected.
+// Logger is also option; if it is provided warning messages may be logged.
+func GenerateManifest(files []string, setFlags []string, force bool, client kube.Client, logger clog.Logger) ([]manifest.ManifestSet, values.Map, error) {
+	// First, compute our final configuration input. This will be in the form of an IstioOperator, but as an unstructured values.Map.
+	// This allows safe access to get/fetch values dynamically, and avoids issues are typing and whether we should emit empty fields.
+	merged, err := MergeInputs(files, setFlags, client)
+	if err != nil {
+		return nil, nil, fmt.Errorf("merge inputs: %v", err)
+	}
+	// Validate the config. This can emit warnings to the logger. If force is set, errors will be logged as warnings but not returned.
+	if err := validateIstioOperator(merged, logger, force); err != nil {
+		return nil, nil, err
+	}
+	// After validation, apply any unvalidatedValues they may have set.
+	if unvalidatedValues, _ := merged.GetPathMap("spec.unvalidatedValues"); unvalidatedValues != nil {
+		merged.MergeFrom(values.MakeMap(unvalidatedValues, "spec", "values"))
+	}
+
+	// Render each component
+	var allManifests []manifest.ManifestSet
+	var chartWarnings util.Errors
+	for _, comp := range component.AllComponents {
+		specs, err := comp.Get(merged)
+		if err != nil {
+			return nil, nil, fmt.Errorf("get component %v: %v", comp.UserFacingName, err)
+		}
+		for _, spec := range specs {
+			// Each component may get a different view of the values; modify them as needed (with a copy)
+			compVals := applyComponentValuesToHelmValues(comp, spec, merged)
+			// Render the chart
+			rendered, warnings, err := helm.Render(spec.Namespace, comp.HelmSubdir, compVals)
+			if err != nil {
+				return nil, nil, fmt.Errorf("helm render: %v", err)
+			}
+			chartWarnings = util.AppendErrs(chartWarnings, warnings)
+			// IstioOperator has a variety of processing steps that are done *after* Helm, such as patching. Apply any of these steps.
+			finalized, err := postProcess(comp, spec, rendered, compVals)
+			if err != nil {
+				return nil, nil, fmt.Errorf("post processing: %v", err)
+			}
+			allManifests = append(allManifests, manifest.ManifestSet{
+				Component: comp.UserFacingName,
+				Manifests: finalized,
+			})
+		}
+	}
+
+	// Log any warnings we got from the charts
+	if logger != nil {
+		for _, w := range chartWarnings {
+			logger.LogAndErrorf("%s %v", "❗", w)
+		}
+	}
+	return allManifests, merged, nil
+}
+
+// applyComponentValuesToHelmValues translates a generic values set into a component-specific one
+func applyComponentValuesToHelmValues(comp component.Component, spec apis.GatewayComponentSpec, merged values.Map) values.Map {
```

Comment:
I'd still like to get us to a point where this function and functions like it can be deleted and we aren't doing _any_ "translation", since there's nothing here that Helm can't do out of the box, or that we need to do differently than Helm (outside of back compat concerns with our own existing weirdness).

(this is not a comment on this PR, which is a big improvement, just a general "I want a pink pony for my birthday" comment)